app/controllers/api/users_controller.rb

```diff
@@ -1,7 +1,41 @@
+require "bcrypt"
+
 class Api::UsersController < Api::ApplicationController
+  wrap_parameters format: [:json]
+  before_action -> { doorkeeper_authorize! :account_data_read }, only: [:verify_pin]
   include Api::PublicApiExposingController
 
   def present
     @users = Presence.present_users
   end
+
+  def phone_number_token
+    secret = Rails.application.secrets.PHONE_TOKEN_SECRET
+    return head :internal_server_error if !secret
+
+    phone_app_id = Rails.application.secrets.PHONE_OAUTH_APP_ID
+    return head :internal_server_error if !phone_app_id
+
+    return head :bad_request if !params[:phone_number]
+    return head :forbidden if params[:secret] != secret
+
+    @user = User.joins(:phone_numbers).where(phone_numbers: {phone_number: params[:phone_number]}).limit(1).first
+    @auth_token = Doorkeeper::AccessToken.find_or_create_for(
+      application: Doorkeeper::Application.find(phone_app_id),
+      resource_owner: @user.id,
+      expires_in: 600,
+      scopes: "door_control account_data_read"
+    )
+    puts "token #{@auth_token.plaintext_token}"
```

Comment:
@lz1irq why do you have print here?

Reply:
Because I forgot to remove it after debugging...